lib/util.js

```diff
@@ -346,7 +346,7 @@ function formatValue(ctx, value, recurseTimes) {
   // Provide a hook for user-specified inspect functions.
   // Check that value is an object with an inspect function on it
   if (ctx.customInspect && value) {
-    const maybeCustomInspect = value[customInspectSymbol] || value.inspect;
+    const maybeCustomInspect = value[customInspectSymbol] || ((typeof value === 'object' && 'inspect' in value) && value.inspect);
```

Comment:
Also can use `Object.hasOwnProperty` that works (in the always false sense) on non-objects